contracts/amm/AmmSwapsLens.sol

```diff
@@ -0,0 +1,391 @@
+                    } else {
+                        if (
+                            iporSwap.endTimestamp - assetConfiguration.timeBeforeMaturityAllowedToCloseSwapByCommunity >
+                            closeTimestamp
+                        ) {
+                            return 4;
+                        }
+                    }
+                }
+            }
+        }
+
+        return 0;
+    }
+
+    function _isLiquidator(address liquidator) internal view returns (bool) {
+        return liquidator0 == liquidator;
+    }
+
+    function getSwapsPayFixed(
+        address asset,
+        address account,
+        uint256 offset,
+        uint256 chunkSize
+    ) external view override returns (uint256 totalCount, IAmmSwapsLens.IporSwap[] memory swaps) {
+        AssetConfiguration memory assetConfiguration = _getAssetConfiguration(asset);
+        IMiltonStorage miltonStorage = IMiltonStorage(assetConfiguration.miltonStorage);
+        (uint256 count, uint256[] memory swapIds) = miltonStorage.getSwapPayFixedIds(account, offset, chunkSize);
+        return (count, _mapSwapsPayFixed(assetConfiguration, swapIds));
+    }
+
+    function getSwapsReceiveFixed(
+        address asset,
+        address account,
+        uint256 offset,
+        uint256 chunkSize
+    ) external view override returns (uint256 totalCount, IAmmSwapsLens.IporSwap[] memory swaps) {
+        AssetConfiguration memory assetConfiguration = _getAssetConfiguration(asset);
+        IMiltonStorage miltonStorage = IMiltonStorage(assetConfiguration.miltonStorage);
+        (uint256 count, uint256[] memory swapIds) = miltonStorage.getSwapReceiveFixedIds(account, offset, chunkSize);
+        return (count, _mapSwapsReceiveFixed(assetConfiguration, swapIds));
+    }
+
+    function getMySwapsPayFixed(
+        address asset,
+        uint256 offset,
+        uint256 chunkSize
+    ) external view override returns (uint256 totalCount, IAmmSwapsLens.IporSwap[] memory swaps) {
+        AssetConfiguration memory assetConfiguration = _getAssetConfiguration(asset);
+        IMiltonStorage miltonStorage = IMiltonStorage(assetConfiguration.miltonStorage);
+        (uint256 count, uint256[] memory swapIds) = miltonStorage.getSwapPayFixedIds(msg.sender, offset, chunkSize);
+        return (count, _mapSwapsPayFixed(assetConfiguration, swapIds));
+    }
+
+    function getMySwapsReceiveFixed(
+        address asset,
+        uint256 offset,
+        uint256 chunkSize
+    ) external view override returns (uint256 totalCount, IAmmSwapsLens.IporSwap[] memory swaps) {
+        AssetConfiguration memory assetConfiguration = _getAssetConfiguration(asset);
+        IMiltonStorage miltonStorage = IMiltonStorage(assetConfiguration.miltonStorage);
+        (uint256 count, uint256[] memory swapIds) = miltonStorage.getSwapReceiveFixedIds(msg.sender, offset, chunkSize);
+        return (count, _mapSwapsReceiveFixed(assetConfiguration, swapIds));
+    }
+
+    function getMySwaps(
```

Comment:
remove getMySwap - we should not use msg.sender inside lens method

Reply:
removed